Refactor `dbt-sqlite` tests to use `dbt-postgres`

pankajastro opened the pull request:
closes: https://github.com/astronomer/astronomer-cosmos/issues/1341

previously ran SQLite tests with dbt-core version 1.4, which has reached its end of life. Recently, these tests have started failing. Since we rely on the repo: https://github.com/codeforkjeff/dbt-sqlite/ for running the tests, and it is no longer actively maintained, this PR performs the necessary cleanup to address the issue

- Point the sqlite test to use Postgres
- Remove CI workflow job: Run-Integration-Tests-Sqlite
- Remove hatch target:  hatch run tests.py3.12-2.10:test-integration-sqlite-setup and  hatch run tests.py3.12-2.10:test-integration-sqlite
- Remove pytest marker `sqlite`
- Remove dbt project simple
- Update manifest file

pankajastro commented on .github/workflows/test.yml:
```diff
@@ -2,7 +2,7 @@ name: test
 on:
   push: # Run on pushes to the default branch
-    branches: [main]
+    branches: [main, replace-sqlite-to-pg]
```
```suggestion
    branches: [main]
```

tatiana reviewed (commented):
Thanks for working on this, @pankajastro , it will be great to no longer depend on an adaptor that stopped being maintained.

Are the example DAGs running as expected?

I had the impression the SQLite tests relied on pre-populated data in the database. Are they running as expected? I was surprised no test is failing - I thought we had an example DAG that relied on this as well.

pankajastro commented:
> Thanks for working on this, @pankajastro , it will be great to no longer depend on an adaptor that stopped being maintained.
> 
> Are the example DAGs running as expected?
> 
> I had the impression the SQLite tests relied on pre-populated data in the database. Are they running as expected? I was surprised no test is failing - I thought we had an example DAG that relied on this as well.

right, I think that particular DAG got ignored because of

```python
if DBT_VERSION >= Version("1.5.0"):
            file.writelines(["example_cosmos_sources.py\n"])
``` 
I'll fix it

pankajastro commented:
> Thanks for working on this, @pankajastro , it will be great to no longer depend on an adaptor that stopped being maintained.
> 
> Are the example DAGs running as expected?
> 
> I had the impression the SQLite tests relied on pre-populated data in the database. Are they running as expected? I was surprised no test is failing - I thought we had an example DAG that relied on this as well.

It was indeed not running. 
But now running: https://github.com/astronomer/astronomer-cosmos/actions/runs/12251360416/job/34175888338?pr=1366#step:7:395
post your previous review, I have removed the dbt project simple from repo since it was getting used for sql test only and updated the dbt project jaffle_shop to make this example and other test work

tatiana commented on dev/dags/dbt/jaffle_shop/models/orders.sql:
```diff
@@ -1,3 +1,9 @@
+{{ config(
+    materialized='view',
+    alias=var('animation_alias', 'orders')
```
Why are we naming an alias to `orders` as `animation_alias`?

tatiana replied:
Makes sense, shall we rename `animation_alias` to `orders_alias`?

pankajastro replied:
Fix in: https://github.com/astronomer/astronomer-cosmos/pull/1366/commits/8fd03dc8afda10a51b3185c5c5f744b4bf9fe28d

tatiana commented on dev/dags/dbt/jaffle_shop/profiles.yml:
```diff
@@ -16,7 +16,7 @@ postgres_profile:
     dev:
       type: postgres
-      dbname: "{{ env_var('POSTGRES_DATABASE') }}"
+      dbname: "{{ env_var('POSTGRES_DB') }}"
```
Why did we change this?

pankajastro replied:
I pointed out the SQLite test to use this profile, but ENV `POSTGRES_DATABASE` is not set in GitHub action, so I just renamed it the env name is already available in CI. But do let me know if it could have any side effect

tatiana replied:
Thanks for the explanation, @pankajastro , I was confused - but it makes sense!

tatiana reviewed (approved):
Thanks for refactoring and restoring these tests, @pankajastro !

It's a shame https://github.com/codeforkjeff/dbt-sqlite hasn't been kept up-to-date for over a year now. It doesn't make sense for us to rely on this adaptor. We can review this decision in the future if circumstances change.
